octoprint_mrbeam/static/js/convert.js

```diff
@@ -616,25 +616,56 @@ $(function () {
             self._update_job_summary();
         });
 
+        self.showMaterialCompatibility = function () {
+            let customMaterialsLaserModels = [];
+            let customMaterials = self.custom_materials();
+            // Get all laser models used while saving custom materials
+            for (let materialKey in customMaterials) {
+                if (customMaterials.hasOwnProperty(materialKey)) {
+                    let m = customMaterials[materialKey];
+                    if (('laser_model' in m) && !customMaterialsLaserModels.includes(m.laser_model)) {
+                        customMaterialsLaserModels.push(m.laser_model);
+                    } else if (!('laser_model' in m) &&
+                        ('laser_type' in m) &&
+                        m.laser_type === 'MrBeamII-1.0' &&
+                        !customMaterialsLaserModels.includes('0')) {
+                        customMaterialsLaserModels.push('0');
+                    }
+                }
+            }
+            return customMaterialsLaserModels.length > 1 ||
+                (customMaterialsLaserModels.length === 1 &&
+                    customMaterialsLaserModels[0] !== MRBEAM_LASER_HEAD_MODEL);
```

Comment:
I don't know how you'd do it in JS, but I feel like this is more transparent / simple ? (python flavour)
```python
return any([ laser != my_laserhead for laser in customMaterialsLaserModels])
```